src/i_winmusic.c

```diff
@@ -1534,7 +1539,7 @@ static void I_WIN_PauseSong(void *handle)
         return;
     }
 
-    paused = true;
+    music_state = stop_sound;
```

Comment:
If the main thread can modify `music_state` before initial playback has finished or whenever it's inside the switch statement, then there's no way to ensure the correct order of execution. You can try to guard with `if (music_state != initial_playback)` inside the pause and resume functions but that's still not a complete solution.

Reply:
Okay, how about just pause states? https://github.com/fabiangreffrath/woof/pull/1114/commits/bc44667c6794fd7c021fb47e3b403f0100423e70
BTW, ZMusic has something similar: https://github.com/ZDoom/ZMusic/blob/fc30a8f1e4350dba68e23bb702dfa24df3fe525e/source/zmusic/musinfo.h#L36-L41

Reply:
Looks good. Let's flush that buffer immediately to be safe.
```diff
    if (music_state == stopped)
    {
        StopSound();
+       StreamOut();
        music_state = paused;
+       return;
    }
```

Reply:
Done in https://github.com/fabiangreffrath/woof/pull/1114/commits/c1a772c20c805bedd446ebea30967aa42a9132a3